src/main/scala/cognite/spark/v1/SyncCursorCallback.scala

```diff
@@ -0,0 +1,59 @@
+package cognite.spark.v1
+
+import cats.effect.IO
+import com.cognite.sdk.scala.sttp.{GzipBackend, RetryingBackend}
+import io.circe.Codec
+import io.circe.generic.semiauto.deriveCodec
+import scala.concurrent.duration._
+import sttp.client3.{SttpBackend, basicRequest}
+import sttp.client3.asynchttpclient.SttpClientBackendFactory
+import sttp.client3.asynchttpclient.cats.AsyncHttpClientCatsBackend
+import sttp.client3.circe.{asJson, circeBodySerializer}
+import sttp.model.{MediaType, Uri}
+
+case class IncrementalCursor(name: String, value: String)
+case class IncrementalCursorResponse(name: String, value: String, updated: Boolean)
+
+object SyncCursorCallback {
+  @transient private lazy val sttpBackend: SttpBackend[IO, Any] =
+    new GzipBackend[IO, Any](AsyncHttpClientCatsBackend.usingClient(SttpClientBackendFactory.create()))
+
+  @transient private val retryingBackend = new RetryingBackend[IO, Any](
+    sttpBackend,
+    maxRetries = 5,
+    initialRetryDelay = 200.milliseconds,
+    maxRetryDelay = 5.seconds
+  )
+
+  implicit val incrementalCursorCodec: Codec[IncrementalCursor] = deriveCodec[IncrementalCursor]
+  implicit val incrementalCursorResponseCodec: Codec[IncrementalCursorResponse] =
+    deriveCodec[IncrementalCursorResponse]
+
+  def lastCursorCallback(
+      callbackUrl: String,
+      cursorName: String,
+      cursorValue: String,
+      jobId: String): IO[IncrementalCursorResponse] = {
+    val uri = Uri.parse(s"$callbackUrl/$jobId") match {
+      case Left(value) => throw new IllegalArgumentException(s"Failed to parse URI '$value'")
```

Comment:
here we should decide what to do with uri parse errors
- burn -> `IO.raiseError(exception here)` or `IO.fromEither(uriparse.left.map(string to exception))` and then `uriio.flatMap(uri => basicRequest...)` or `for {uri <- urlio; response <- basicrequest... } yield ...`
- lift to `IO[Either[someerrortype, IncrementalCursorResponse]]` to present it as if it was a request/response error

Reply:
another option is to parse/filter the callbackUri early and have it as URI type already

Reply:
The URL is coming from the dataframe setup, though, so I would say this isn't really that much needed.

Reply:
If there's a error on putting that from dataframe setup side, it should be made visible somehow